plugins/by-name/openscad/default.nix

```diff
@@ -1,81 +1,93 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
-with lib;
 let
-  defaultFuzzyFinder = "skim";
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib.nixvim.vim-plugin) mkSettingsOptionDescription applyPrefixToAttrs;
+
+  name = "openscad";
+  globalPrefix = "openscad_";
 in
-{
-  options.plugins.openscad = {
-    enable = mkEnableOption "openscad.nvim, a plugin to manage OpenSCAD files";
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  inherit name;
+  packPathName = "openscad.nvim";
+  package = "openscad-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  # TODO: Added 2024-12-17; remove after 25.05
+  optionsRenamedToSettings = import ./renamed-options.nix;
```

Comment:
Shall we make `optionsRenamedToSettings` auto-import if needed?

```
fn = maybeList:
if isPath maybeList then fn (import maybeList) else # do thing...
```

Reply:
No, to me that's a bit too overkilled. It would hide even more semantics. I would prefer to KISS in this case.

Reply:
It's pretty common practice in nixpkgs for utils like `callPackage` to accept _either_ a value or a path that contains a value when imported. But no issues with not doing that.